repository/src/test/java/org/alfresco/repo/event2/EventGeneratorTest.java

```diff
@@ -0,0 +1,221 @@
+    private static final boolean DEBUG = true;
+
+    private static final String EVENT2_TOPIC_NAME = "alfresco.repo.event2";
+
+    private ActiveMQConnection connection;
+    private ObjectMapper objectMapper;
+    protected List<RepoEvent<?>> receivedEvents;
+
+    @Before
+    public void startupTopicListener() throws Exception
+    {
+        ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory("tcp://localhost:61616");
+        connection = (ActiveMQConnection) connectionFactory.createConnection();
+        connection.start();
+
+        Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        Destination destination = session.createTopic(EVENT2_TOPIC_NAME);
+        MessageConsumer consumer = session.createConsumer(destination);
+
+        objectMapper = ObjectMapperFactory.createInstance();
+        receivedEvents = Collections.synchronizedList(new ArrayList<>());
+        consumer.setMessageListener(new MessageListener()
+        {
+            @Override
+            public void onMessage(Message message)
+            {
+                String text = getText(message);
+                RepoEvent<?> event = toRepoEvent(text);
+
+                if (DEBUG)
+                {
+                    System.err.println("Received message " + message + "\n" + text + "\n" + event);
+                }
+
+                receivedEvents.add(event);
+            }
+
+            private RepoEvent<?> toRepoEvent(String json)
+            {
+                try
+                {
+                    return objectMapper.readValue(json, RepoEvent.class);
+                }
+                catch (Exception e)
+                {
+                    e.printStackTrace();
+                    return null;
+                }
+            }
+        });
+
+        if (DEBUG) System.err.println("Now actively listening on topic " + EVENT2_TOPIC_NAME);
+    }
+
+    @After
+    public void shutdownTopicListener() throws Exception
+    {
+        connection.close();
+        connection = null;
+    }
+
+    @Test
+    public void shouldReceiveEvent2EventsOnNodeCreation() throws Exception
+    {
+        createNode(ContentModel.TYPE_CONTENT);
+
+        int i = 100;
+        while (--i > 0)
+        {
+            Thread.sleep(55l);
+            if (receivedEvents.size() == 1)
+                break;
+        }
+
+        assertFalse("No messages were received!", receivedEvents.isEmpty());
+
+        RepoEvent<?> sent = getRepoEvent(1);
+        RepoEvent<?> received = receivedEvents.get(0);
+
+        assertEquals("Events are different!", sent, received);
+    }
+
+    private static String getText(Message message)
+    {
+        try
+        {
+            ActiveMQTextMessage am = (ActiveMQTextMessage) message;
+            return am.getText();
+        }
+        catch (JMSException e)
+        {
+            return null;
+        }
+    }
+
+    // a simple main to investigate he contents of the local broker
```

Comment:
a small typo "...to investigate **t**he contents..."

Reply:
Sorted out.